pkg/apis/core/v1/storagecluster.go

```diff
@@ -558,7 +559,8 @@ type PrometheusSpec struct {
 	// RemoteWriteEndpoint specifies the remote write endpoint
 	RemoteWriteEndpoint string `json:"remoteWriteEndpoint,omitempty"`
 	// AlertManager spec for configuring alert manager
-	AlertManager *AlertManagerSpec `json:"alertManager,omitempty"`
+	AlertManager                *AlertManagerSpec `json:"alertManager,omitempty"`
+	monitoringv1.PrometheusSpec `json:",inline"`
```

Comment:
I feel we should not directly expose the entire prometheus spec. It adds too much dependency on a third party and we lose control over the fields a user can set. Also, as the vendor changes, the spec would change based on what prometheus has added. Especially when we start automatically generating the schema for the CRDs.

Reply:
right now we only expose securityContext and resources in stc crd. ideally we can expose everything prometheus supports if we want, it actually has a lot of good features. for example in the ticket there is also request for adding non root file system support, prometheus has an issue to fix it, if in future it's fixed we will just use it

Reply:
What we are doing here is blocking the other specs only through the CRD yaml schema, but the go code still has everything. If you want to expose more fields we can still add direct fields here, right? Anyway you would need a code/schema change.
Also, this will affect PXOne which generates a proto from this Go API. The proto will have the entire prometheus spec.